### PR TITLE
Ensure that we can use aliases genearted in compute on later stages

### PR DESCRIPTION
**Pull request by kosinsky**
Ensure that parses can bind properties introduced in compute in orderby or filter

### CODE REVIEW

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs [hunk 1]
```diff
@@ -51,6 +53,13 @@ public ApplyClause(IList<TransformationNode> transformations)
 
                     break;
                 }
+                else if (transformations[i].Kind == TransformationNodeKind.Compute)
+                {
+                    if (lastAggregateExpressions == null)
```
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs]: >lastAggregateExpressions [](start = 24, length = 24)

`lastComputeExpressions`?

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs [hunk 2]
```diff
@@ -51,6 +53,13 @@ public ApplyClause(IList<TransformationNode> transformations)
 
                     break;
                 }
+                else if (transformations[i].Kind == TransformationNodeKind.Compute)
+                {
+                    if (lastAggregateExpressions == null)
+                    {
+                        lastComputeExpressions = (transformations[i] as ComputeTransformationNode).ComputeClause.ComputedItems;
```
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs]: > lastComputeExpressions = (transformations[i] as ComputeTransformationNode).ComputeClause.ComputedItems; [](start = 23, length = 104)

In line 41 there is a break after lastAggregateExprssions is set. Do you want to do the same thing here? #Resolved

Comment (StanislawSwierc) [reply src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs]: I see below that probably no.

---
In reply to: [173582650](https://github.com/kosinsky/odata.net/pull/9#discussion_r173582650) [](ancestors = 173582650)

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs (hunk 1 above)
Comment (kosinsky) [reply src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs]: No, lastAggregateExpressions to support scenario from ReferenceComputeAliasCreatedBeforeAggrageteThrows

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs [hunk 3]
```diff
@@ -72,12 +81,21 @@ internal string GetContextUri()
...
 
-            return null;
+            List<string> result = new List<string>();
+            if (lastAggregateExpressions != null)
+            {
+                result.AddRange(lastAggregateExpressions.Select(statement => statement.Alias));
+            }
+            if (lastComputeExpressions != null)
+            {
+                result.AddRange(lastComputeExpressions.Select(statement => statement.Alias));
```
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs:94]: >result.AddRange(lastComputeExpressions.Select(statement => statement.Alias)); [](start = 16, length = 77)

Can there be duplicates/conflicts when you do 
```
groupby((Type), aggregate($count as Count, Value with sum as Value))/compute(Value div Count as Value)
```

File: test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs [hunk 4]
```diff
@@ -1142,6 +1142,22 @@ public void AggregatedPropertyTreatedAsOpenPropertyInOrderBy()
...
 
+        [Fact]
+        public void ComputedPropertyTreatedAsOpenPropertyInOrderBy()
+        {
+            var odataQueryOptionParser = new ODataQueryOptionParser(HardCodedTestModel.TestModel,
+                HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet(),
+                new Dictionary<string, string>()
+                {
+                    {"$orderby", "DoubleTotal asc"},
+                    {"$apply", "aggregate(FavoriteNumber with sum as Total)/compute(Total mul 2 as DoubleTotal)"}
```
Comment (StanislawSwierc) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs:1153]: >DoubleTotal [](start = 99, length = 11)

What would happen if you renamed it to `Total`?

**Review by StanislawSwierc — CHANGES_REQUESTED** [review]
:clock1:

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs (hunk 2 above)
Comment (StanislawSwierc) [inline src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs]: >lastComputeExpressions [](start = 24, length = 22)

As discussed you may need to keep appending to this collection to cover the case of a series of compute clauses.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs (hunk 4 above)
Comment (kosinsky) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs:1153]: Good question and we have the same issue with aggregate. Not going to fix it as part of this PR

File: src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs (hunk 3 above)
Comment (kosinsky) [reply src/Microsoft.OData.Core/UriParser/Aggregation/ApplyClause.cs:94]: Good question and we have the same issue with aggregate. Not going to fix it as part of this PR

**Review by StanislawSwierc — APPROVED** [review]
:shipit:

**Pull request merged**